elastic/integrations
# Sql generic datastream ver1

## What does this PR do?

This PR contributes the performance data stream for Microsoft sqlserver which fetches information from what’s commonly known as [Performance Counters](https://docs.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-os-performance-counters-transact-sql?view=sql-server-2017) in MSSQL.

- Added 1 data stream (performance) under microsoft_sqlserver integration package
- Added data collection logic for the data streams.
- Added Fields metadata in the appropriate yaml files.
- Added dashboards and visualizations.

## Checklist


## How to test this PR locally

- Install the MSSQL 
- Go to Integration UI in Kibana
- Add the Microsoft SQL Server Integration
- Enroll the new elastic-agent
- In the asset -> Performance dashboard you should be able to see the default visualization.

## Related issues

TBD


## Screenshots

![mssql-perf-dashboard](https://user-images.githubusercontent.com/97392538/169289189-474e6eb7-241c-407d-a312-a13f92063d77.png)
